lib/model/avatar_url.dart

```diff
@@ -0,0 +1,148 @@
+import 'dart:convert';
+
+import 'package:crypto/crypto.dart';
+import 'package:path/path.dart';
+
+import '../log.dart';
+
+/// Pixel dimensions of different size choices we have (they're all
+/// square) when requesting an uploaded avatar.
+/// DEFAULT_AVATAR_SIZE in zerver/lib/upload.py.
+const int defaultUploadSizePx = 100;
+
+/// MEDIUM_AVATAR_SIZE in zerver/lib/upload.py.
+const int mediumUploadSizePx = 500;
+
+abstract class AvatarURL {
```

Comment:
nit: follow [camel-casing rules](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#capitalize-identifiers-consistent-with-their-spelling):

```suggestion
abstract class AvatarUrl {
```

I know the corresponding names in zulip-mobile are spelled like "AvatarURL", but we're being more consistent in zulip-flutter.

Reply:
[c6715ea](https://github.com/zulip/zulip-flutter/pull/534/commits/c6715ea2f8091ddfa06162cc41e32035ea7f72a4)